### ISSUE TITLE
Pass down options to Flatpickr on the DatePicker component

### ISSUE THREAD

**Issue by enricoberti** [carbon-design-system/carbon#5713]
### Summary

It would be nice to be able to pass down to Flatpickr itself some options that are you can't set on the flatpickr instance itself (ie. `inline: true`)

In the React DatePicker component it's a matter of adding a prop and then spreading it when initing flatpickr

### Justification

We do have the need to have an inline/static picker on carbon-addons-iot for [this story](https://github.com/IBM/carbon-addons-iot-react/issues/854)

As it right now, we are reparenting the calendar DOM into our component but this is far from the best option to do it.

### Specific timeline issues / requests

For now we are going with the workaround but it would be nice to have soon :)

### Available extra resources

I can contribute back myself but I need some support from the Carbon team given that adding a new prop to DatePicker would break semver.

Thanks!

**Comment by emyarod** [comment]
I believe the restriction on flatpickr options configuration is intentional, specifically around the `inline` prop

cc @carbon-design-system/design on whether or not inline date picker is a valid variant within the design system

**Comment by aagonzales** [comment]
It was originally intentional to not include the inline format for datepicker. I think if we wanted to include it, it would need some visual exploration because of the calendar icon. This topic needs some investigation from the design perspective before we can determine if we should do it or not.

**Comment by enricoberti** [comment]
hi @aagonzales!  design-wise you can also ping @lukefirth and @cynthiaz who worked on https://github.com/IBM/carbon-addons-iot-react/issues/854

**Comment by jinhei** [comment]
Along the same lines, it would be nice to be able to pass down `disabled` and `enabled` flags to flatpickr, which we're currently unable to do. Our use case is that only certain days should be selectable based on availability.

See [flatpickr example documentation](https://flatpickr.js.org/examples/#disabling-specific-dates).

**Comment by davidicus** [comment]
Any discussion still being had on this?
